### ISSUE TITLE
[BUG] get_json_object cannot handle ints or boolean values

### ISSUE THREAD

**Issue by revans2** [NVIDIA/spark-rapids#10700]
**Describe the bug**
The legacy version of GetJsonObject works just fine, but it looks like the new version does not handle normalization/validation properly. I am also really concerned about null

**Steps/Code to reproduce bug**
```
Seq("{'a': true}", "{'a': false}", "{'a': null}").toDF("s").repartition(1).selectExpr("s", "get_json_object(s, '$.a')").collect.foreach(println)
(0 until 100).map(i => "{'a': 1" + ("0" * i) + "}").toDF("s").repartition(1).selectExpr("s", "get_json_object(s, '$.a')").collect.foreach(println)
```

return nulls for all of the values instead of matching what Spark does, or the legacy version.

Our tests appear to really be lacking in this area.

**Comment by revans2** [comment]
So it looks like most of this has been fixed in 24.06 after the upmerge went in. I will retest things

**Comment by res-life** [comment]
Has been fixed after this upmerge from branch 24.04 to 24.06: 
[Fix merge from 24.04 to 24.06](https://github.com/NVIDIA/spark-rapids-jni/pull/1965)

Now both branch (24.04 and 24.06) are fixed.

I'll add the test cases in this issue into Spark-Rapids.
